examples/datasets/front_3d_object_sampling/main.py

```diff
@@ -0,0 +1,154 @@
+
+# Select the objects, where other objects should be sampled on
+left_objects = []
+for obj in room_objs:
+    if "table" in obj.get_name().lower() or "desk" in obj.get_name().lower():
+        left_objects.append(obj)
+
+for obj in left_objects:
+    # The loop starts with and UndoAfterExecution in order to restore the camera positions for each object
+    with bproc.python.utility.Utility.Utility.UndoAfterExecution():
+        # Select the surfaces, where the object should be sampled on
+        droppable_surface = bproc.object.slice_faces_with_normals([obj])
+        if droppable_surface is not None and len(droppable_surface) == 1:
+            surface_obj = droppable_surface[0]
+        else:
+            continue
+
+        surface_height_z = np.mean(surface_obj.get_bound_box(), axis=0)[2]
+        def sample_pose(obj: bproc.types.MeshObject):
+            # Sample the spheres location above the surface
+            obj.set_location(bproc.sampler.upper_region(
+                objects_to_sample_on=[surface_obj],
+                min_height=1,
+                max_height=4,
+                use_ray_trace_check=False
+            ))
+            #Randomized rotation of the sampled object
+            obj.set_rotation_euler(np.random.uniform([0, 0, 0], [np.pi * 2, np.pi * 2, np.pi * 2]))
+
+        # Load the object, which should be sampled on the surface
+        sampling_obj = bproc.loader.load_blend(args.treed_obj_path)
+        dropped_object_list = bproc.object.sample_poses_on_surface(sampling_obj, surface_obj, sample_pose,
+                                                                   min_distance=0.1, max_distance=10,
+                                                                   check_all_bb_corners_over_surface=False)
+        if not dropped_object_list:
+            print("Dropping of the object failed")
+            continue
+
+        # dropped_objects.append(dropped_object_list)
+        # Enable physics for spheres (active) and the surface (passive)
+        for dropped_object in dropped_object_list:
+            dropped_object.enable_rigidbody(True)
+        surface_obj.enable_rigidbody(False)
+
+        # Run the physics simulation
+        bproc.object.simulate_physics_and_fix_final_poses(min_simulation_time=2, max_simulation_time=4,
+                                                          check_object_interval=1)
+
+        def conv_to_homogen(x):
+            x = x.co.to_4d()
+            x[3] = 1.0
+            return x
+
+        # join surface objects again
+        bpy.ops.object.select_all(action='DESELECT')
+        obj.select()
+        surface_obj.select()
+        bpy.context.view_layer.objects.active = surface_obj.blender_obj
+        bpy.ops.object.join()
+        bpy.ops.object.select_all(action='DESELECT')
+
+        min_coord_z = np.min([dropped_object.get_local2world_mat() @ conv_to_homogen(vert) for vert in
+                              dropped_object.blender_obj.data.vertices], axis=0)[2]
```

Comment:
Why do you not just use the min_z of the object's bounding box? Wouldn't this be equivalent?